runtime/vm/resolvesupport.cpp

```diff
@@ -1979,6 +2084,56 @@ resolveMethodHandleRef(J9VMThread *vmThread, J9ConstantPool *ramCP, UDATA cpInde
 
 }
 
+j9object_t
+resolveInvokeHandle(J9VMThread *vmThread, J9ConstantPool *ramCP, UDATA cpIndex, UDATA resolveFlags)
+{
+	j9object_t result = NULL;
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+	bool canRunJavaCode = J9_ARE_NO_BITS_SET(resolveFlags, J9_RESOLVE_FLAG_JIT_COMPILE_TIME | J9_RESOLVE_FLAG_REDEFINE_CLASS);
+	J9RAMMethodRef *ramCPEntry = (J9RAMMethodRef*)ramCP + cpIndex;
+	UDATA invokeCacheIndex = ramCPEntry->methodIndexAndArgCount >> 8;
+	J9Class *ramClass = J9_CLASS_FROM_CP(ramCP);
+	j9object_t *invokeCache = ramClass->invokeCache + invokeCacheIndex;
+	result = *invokeCache;
+
+	if (NULL != result) {
+		return result;
+	} else if (!canRunJavaCode) {
+		return NULL;
```

Comment:
The code can be rewritten to have only one return statement. This will allow entry and exit tracepoints to be correctly located.

Reply:
fixed